src/coordination/coordinator_instance.cpp

```diff
@@ -539,9 +548,39 @@ auto CoordinatorInstance::SetReplicationInstanceToMain(std::string_view instance
   return SetInstanceToMainCoordinatorStatus::SUCCESS;
 }
 
+auto CoordinatorInstance::HasBecomeLeader() -> bool {
+  if (!raft_state_.RequestLeadership()) {
+    spdlog::error("Request leadership denied");
+    return false;
+  }
+
+  using namespace std::chrono_literals;
+  auto lock = std::unique_lock{leader_mutex_};
+  auto expontential_backoff{1.2};
+  auto waiting_period = 1s;
+  auto maybe_stop = utils::ResettableCounter<5>();
+  do {
+    spdlog::trace("Waiting for coordinator to become leader.");
+    if (!become_leader_.wait_for(lock, waiting_period, [this]() { return leader_set_up_.load(); })) {
+      waiting_period *= expontential_backoff;
+    }
+  } while (!maybe_stop() && !leader_set_up_);
+
+  return leader_set_up_;
+}
```

Comment:
I understand this, we are waiting for callback to be executed to make sure we are the leader, right?